roles/smallfile-bench/tasks/main.yml

```diff
@@ -0,0 +1,92 @@
+---
+- set_fact:
+    smallfile_path: /mnt/pvc
+  when: smallfile.storageclass is defined
+
+- name: Create PVC(s)
+  k8s:
+    definition:
+      kind: PersistentVolumeClaim
+      apiVersion: v1
+      metadata:
+        name: claim"{{ item }}"
+        namespace: "{{ operator_namespace }}"
+        annotations:
+          volume.beta.kubernetes.io/storage-class: "{{smallfile.storageclass}}"
+      spec:
+        accessModes:
+          - ReadWriteOnce
+        resources:
+          requests:
+            storage: "{{smallfile.storagesize}}"
+  with_sequence: start=1 count={{smallfile.clients}}
+  when: smallfile.clients > 0 and smallfile.storageclass is defined and smallfile.storagesize is defined
+
+- name: Generate smallfile test input yaml
+  k8s:
+    definition:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        namespace: "{{ operator_namespace }}"
+        name: smallfile-test
+      data:
+        smallfilejob: "{{ lookup('template', 'smallfilejob.other_parameters.yaml.j2') }}"
+
+- name: Start Run(s) with PVC
+  k8s:
+    definition:
+      kind: Job
+      apiVersion: batch/v1
+      metadata:
+        name: "{{ meta.name }}-smallfile-client-{{ item }}-benchmark"
+        namespace: "{{ operator_namespace }}"
+      spec:
+        ttlSecondsAfterFinished: 600
+        template:
+          metadata:
+            labels:
+              app: smallfile-benchmark
+          spec:
+            containers:
+              - name: benchmark-server
+                securityContext:
+                  runAsUser: 0 #0 means run as root
+                  privileged: true
+                image: "quay.io/nichawla/smallfile_container:testing"
+                command: ["/bin/sh", "-c"]
+                workingDir: /root/smallfile-master/
+                args:
+                  - arr=$(python <<< "print(' '.join({{smallfile.operation}}))");
```

Comment:
just a suggestion but if the args seems to be having a lot of work, maybe you can make this a script and make it the entry point for the container?

Reply:
@chawlanikhil24 what are your thoughts on this?

Reply:
@aakarshg I tried to implement, but it was not a successful attempt. May be we can keep this for future track and meanwhile keep it here. Although, I've move the definition of the pod into template.